src/classes/BGE_DataImportBatchEntry_CTRL.cls

```diff
@@ -346,28 +347,40 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
                 if (dataImportDFR != null) {
 
                     if (UTIL_Permissions.canUpdate(dataImportDFR, false)) {
+
                         Column col = new Column();
+
                         col.label = dataImportDFR.getLabel();
                         col.fieldName = dataImportDFR.getName();
-                        col.readOnly = false;
                         col.defaultValue = field.defaultValue;
                         col.required = field.requiredInEntryForm;
                         col.hide = field.hide;
-                        col.type = UTIL_LightningComponentSupport.getLightningDataType(dataImportDFR.getType().name());
+                        col.type = UTIL_LightningComponentSupport.getLightningDataTableDataType(dataImportDFR.getType().name());
+                        // lookups aren't editable in the grid since we don't have lookup support there
+                        col.readOnly = col.type == 'reference' ? true : false;
                         col.typeAttributes = '{}';
-
                         col.options = BGE_BatchGiftEntry_UTIL.getPicklistOptions(col.fieldName);
 
                         columns.add(col);
 
-                        String targetObject = BDI_DataImportService.getTargetObject(col.fieldName);
-                        if (targetObject == 'Payment') {
-                            targetObject = 'npe01__OppPayment__c';
+                        if (dataImportDFR.getType().name() == 'reference') {
+                            // manually construct EXTRA column for campaign link
```

Comment:
😢

Reply:
I like that Randi's review is at least 50% in emoji form :)